internal/sbom/cyclonedx.go

```diff
@@ -22,18 +22,30 @@ func (c *CycloneDX) Name() string {
 }
 
 func (c *CycloneDX) enumeratePackages(bom *cyclonedx.BOM, callback func(Identifier) error) error {
-	for _, component := range *bom.Components {
-		if component.PackageURL != "" {
-			err := callback(Identifier{
-				PURL: component.PackageURL,
-			})
-			if err != nil {
-				return err
+	// Components can have components, so enumerate them recursively.
+	var enumerateComponents func([]cyclonedx.Component) error
+	enumerateComponents = func(components []cyclonedx.Component) error {
```

Comment:
nit: can we just pull this out to it's own top level method (with `c *CycloneDX`)? 

(we probably could have also done `enumerateComponents := ` instead with the explicit `var`, but a top level function is still better)

Reply:
I've pulled this out, though now `enumeratePackages` looks a bit redundant.

(I couldn't do `enumerateComponents :=` because it needs to be defined for the recursive call)